### ISSUE TITLE
Maintain FPGA Compatibility

### ISSUE THREAD

**Issue by farzamgl** [black-parrot/black-parrot#728]
Currently, some bare-metal tests have issues running on the Xilinx Virtex Utralscale+ board. We have to debug these issues using Vivado's post-synth simulation and add compatibility tests to our regression suite. (example: https://github.com/black-parrot/black-parrot/pull/706)

**Comment by dpetrisko** [comment]
@Shashank-Vijay is working on post-synth vivado sims, not with the Xilinx IPs though. The bug in #706 would have been caught without the IPs, but I can imagine bugs which would not

**Comment by taylor-bsg** [comment]
And to be clear, we are okay with Vivado synthesis and VCS simulation.
Vivado is 10X slower than VCS.

**Comment by dpetrisko** [comment]
>And to be clear, we are okay with Vivado synthesis and VCS simulation.
>Vivado is 10X slower than VCS.

That's interesting! Probably want to support both, but run most regressions with VCS for speed

**Comment by taylor-bsg** [comment]
Just in terms of spending time on things, bigger fish to fry if it takes a
lot of work to get Vivado sim going -- people can use Verilator

**Comment by dpetrisko** [comment]
>Just in terms of spending time on things, bigger fish to fry if it takes a
lot of work to get Vivado sim going -- people can use Verilator

True, although one major use case is students working with just Vivado for FPGA-based classes. In any case, the idea is for the makefile flow to support orthogonal synth and sim by switching out simulation flists, so it shouldn't be too bad.  But point taken, won't spend an inordinate amount of time there

**Comment by dpetrisko** [comment]
>Just in terms of spending time on things, bigger fish to fry if it takes a
lot of work to get Vivado sim going -- people can use Verilator

From talking to Dustin and Jon, not worth supporting vivado sim.  Too many bugs!

**Comment by taylor-bsg** [comment]
Yep. There is one use case though that is not covered by Verilator, which
is if the Xilinx IP blocks are encrypted. But cross that bridge later.

M
